### PR TITLE
dstore: bringing to mca

### PR DESCRIPTION
**Pull request by karasevb**
In progress. Will be fixed:
- [x] ./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:0-2;1:3]"
- [x] ./pmix_test -n 4 --ns-dist 3:1 --fence "[db | 0:;1:]"
- [x] ./pmix_test -n 4 --ns-dist 3:1 --fence "[0:]"
- [x] ./pmix_test -n 4 --ns-dist 3:1 --fence "[b | 0:]"
- [x] ./pmix_test -n 4 --ns-dist 3:1 --fence "[d | 0:]" --noise "[0:0,1]"
- [x] ./pmix_test -n 4 --job-fence -c
- [x] ./pmix_test -n 4 --job-fence
- [x] ./pmix_test -n 2 --test-publish
- [x] ./pmix_test -n 2 --test-spawn
- [x] ./pmix_test -n 2 --test-connect
- [x]  ./pmix_test -n 5 --test-resolve-peers --ns-dist "1:2:2"
- [x] ./pmix_test -n 5 --test-replace 100:0,1,10,50,99
- [x] ./pmix_test -n 5 --test-internal 10

Signed-off-by: Boris Karasev <karasev.b@gmail.com>

### CODE REVIEW

**Comment by rhc54** [comment]
@karasevb Just an FYI. The ```test-resolve-peers``` test is forcing a change to the GDS interface. I should finish it today. It might actually help make your components easier - just a bit of a pain for mine.

**Comment by rhc54** [comment]
I'm going to update the branch, which might cause some conflicts with your commits. However, I don't want to mess you up by merging this pull request before you are ready.

**Comment by karasevb** [comment]
Updated progress status

**Comment by rhc54** [comment]
Just FYI: all tests appear to be working now without dstore. Many thanks for the assist! Let me know when you are ready and I'll pull your changes into my branch

**Comment by karasevb** [comment]
Synchronized with main PR. I'm working for a resolve peers problem.

**Comment by karasevb** [comment]
@rhc54 all tests passed with dstore. I will do the code cleanup and additional checks, then we can merge. I'll let you know when it's ready.

**Comment by rhc54** [comment]
Very good - thank you!

File: src/mca/gds/gds.h [hunk 1]
```diff
@@ -289,10 +289,13 @@ typedef pmix_status_t (*pmix_gds_base_module_add_nspace_fn_t)(const char *nspace
...
 /* define a convenience macro for add_nspace based on peer */
-#define PMIX_GDS_ADD_NSPACE(s, p, n, i, ni)                 \
+#define PMIX_GDS_ADD_NSPACE(s, n, i, ni)                    \
     do {                                                    \
-        pmix_gds_base_module_t *_g = (p)->nptr->compat.gds; \
-        (s) = _g->add_nspace(n, i, ni);                     \
+        pmix_gds_base_active_module_t *_g;                  \
+        PMIX_LIST_FOREACH(_g, &pmix_gds_globals.actives,    \
+                          pmix_gds_base_active_module_t) {  \
+            (s) = _g->module->add_nspace(n, i, ni);         \
```
Comment (rhc54) [inline src/mca/gds/gds.h]: Sorry about that - the comment didn't go where I put it. Anyway, I was just suggesting that you check for NULL  before calling the function

File: src/mca/gds/gds.h [hunk 2]
```diff
@@ -306,10 +309,13 @@ typedef pmix_status_t (*pmix_gds_base_module_add_nspace_fn_t)(const char *nspace
...
 /* define a convenience macro for del_nspace based on peer */
-#define PMIX_GDS_DEL_NSPACE(s, p, n)                        \
+#define PMIX_GDS_DEL_NSPACE(s, n)                           \
     do {                                                    \
-        pmix_gds_base_module_t *_g = p->nptr->compat.gds;   \
-        (s) = _g->del_nspace(n);                            \
+        pmix_gds_base_active_module_t *_g;                  \
+        PMIX_LIST_FOREACH(_g, &pmix_gds_globals.actives,    \
+                          pmix_gds_base_active_module_t) {  \
+            (s) = _g->module->del_nspace(n);                \
```
Comment (rhc54) [inline src/mca/gds/gds.h]: and same here

**Comment by karasevb** [comment]
@rhc54 Thanks, I added the checking for NULL

**Comment by karasevb** [comment]
@rhc54 ready to merge

**Comment by rhc54** [comment]
Very good - thanks! I'm still thinking about your email question. It seems we always need the hash module, and so we may need to move it to the "base" so it is always available. Otherwise, it could be ignored using MCA parameters.

I'll continue to think about it - we have a little time before this can come into master anyway as we don't want it committed until after the v2.0 branch has been created.

**Pull request merged**